builder/amazon/common/ssh_test.go

```diff
@@ -28,15 +28,15 @@ func TestSSHHost(t *testing.T) {
 		wantHost string
 	}{
 		{1, "", false, true, publicDNS},
-		{1, "", true, true, publicDNS},
+		{1, "", true, true, privateIP},
 		{1, "vpc-id", false, true, publicIP},
 		{1, "vpc-id", true, true, privateIP},
 		{2, "", false, true, publicDNS},
-		{2, "", true, true, publicDNS},
+		{2, "", true, true, privateIP},
 		{2, "vpc-id", false, true, publicIP},
 		{2, "vpc-id", true, true, privateIP},
-		{3, "", false, false, ""},
-		{3, "", true, false, ""},
+		{2, "", false, true, publicDNS},
+		{2, "", true, true, privateIP},
```

Comment:
I'm not sure why I couldn't get these to pass with `allowTries=3`. Now they are redundant with line 34 and 35. But I don't want to remove them unless I'm sure I didn't remove a important test case. Please help @tarrall @dpiddy